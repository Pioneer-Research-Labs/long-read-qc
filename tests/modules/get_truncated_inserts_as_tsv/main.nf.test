nextflow_process {

    name "Test Process get_truncated_inserts_as_tsv"
    script "modules/get_truncated_inserts_as_tsv/main.nf"
    process "get_truncated_inserts_as_tsv"

    test("Should generate inserts from truncated flanks"){

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple([ id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/inserts_from_truncated_flanks.fasta'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$outputDir/Gateway_EmptyBarcodedLib1_20241217/truncated_inserts.tsv").match()
        }
    }
}
