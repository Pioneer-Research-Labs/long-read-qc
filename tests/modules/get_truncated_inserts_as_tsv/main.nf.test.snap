{
    "Should generate inserts from truncated flanks": {
        "content": [
            "truncated_inserts.tsv:md5,8a231fb29d28f8ed2d09c30555947cbd"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-21T12:51:54.389857"
    }
}